tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -216,12 +216,14 @@ def raise_ie_not_found():
         log.error(e)
         raise_ie_not_found()
 
-    if ('data_type' in argv and argv.data_type in ['FP16', 'half']) or \
-            ('compress_to_fp16' in argv and argv.compress_to_fp16 is True):
-        argv.data_type = 'FP32'
-        argv.compress_fp16 = True
-    else:
+    # Turn off compression only if it's disabled explicitly by --compress_to_fp16=False or --data_type=FP32.
+    # By default, in all other cases compression is enabled
+    if ('data_type' in argv and argv.data_type in ['FP32', 'float']) or \
+            ('compress_to_fp16' in argv and argv.compress_to_fp16 is False):
```

Comment:
We should check that there is no contradiction between values of these options, i.e. there should not be cases like 
`argv.data_type=FP32` and `argv.compress_to_fp16=True`.